### PR TITLE
Add authorisation to ReplacePatchResponsibleEntities

### PR DESCRIPTION
**Pull request by adamtry**
## [MR-764](https://hackney.atlassian.net/jira/software/c/projects/MR/boards/88?selectedIssue=MR-764)

PUT `patch/{id}/responsibleEntities`
Endpoint needs authorisation to ensure the client user is in the asset admin google group(s) (as on parameter store)

[MR-764]: https://hackney.atlassian.net/browse/MR-764?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: PatchesAndAreasApi/PatchesAndAreasApi.csproj [hunk 1]
```diff
@@ -26,6 +26,7 @@
     <PackageReference Include="AWSXRayRecorder.Core" Version="2.11.1" />
     <PackageReference Include="AWSXRayRecorder.Handlers.AspNetCore" Version="2.8.1" />
     <PackageReference Include="AWSXRayRecorder.Handlers.AwsSdk" Version="2.9.1" />
+    <PackageReference Include="Hackney.Core.Authorization" Version="1.79.0-feat-regenerate-0002" />
```
Comment (humulla) [inline PatchesAndAreasApi/PatchesAndAreasApi.csproj:29]: dont forget to update this nuget version

File: PatchesAndAreasApi.Tests/V1/E2ETests/Stories/ReplacePatchResponsibleEntitiesTests.cs [hunk 2]
```diff
@@ -28,8 +28,15 @@ public ReplacePatchResponsibleEntitiesTests(MockWebApplicationFactory<Startup> a
             _snsFixture = appFactory.SnsFixture;
             _patchFixture = new PatchesFixtures(_dbFixture.DynamoDbContext, _snsFixture.SimpleNotificationService);
             _steps = new ReplacePatchResponsibleEntitiesStep(appFactory.Client);
+
+            // AppContext.SetSwitch("Npgsql.EnableLegacyTimestampBehavior", true);
```
Comment (humulla) [inline PatchesAndAreasApi.Tests/V1/E2ETests/Stories/ReplacePatchResponsibleEntitiesTests.cs:32]: do we need this?

**Review by humulla — APPROVED** [review]

File: PatchesAndAreasApi.Tests/V1/E2ETests/Stories/ReplacePatchResponsibleEntitiesTests.cs (hunk 2 above)
Comment (adamtry) [reply PatchesAndAreasApi.Tests/V1/E2ETests/Stories/ReplacePatchResponsibleEntitiesTests.cs]: Removed in 3433ebb189a94a9d3bb6218e2ea13e0aeb70a9f3

File: PatchesAndAreasApi/PatchesAndAreasApi.csproj (hunk 1 above)
Comment (adamtry) [reply PatchesAndAreasApi/PatchesAndAreasApi.csproj]: Switched to main version in https://github.com/LBHackney-IT/patches-and-areas-api/pull/49/commits/55696bde22562b051469b5e6470c1611f95e3d83

**Pull request merged**